src/actions/user-actions.js

```diff
@@ -327,6 +336,10 @@ export const updateMembershipType = (type) => (dispatch, getState) => {
     authErrorHandler
   )(params)(dispatch)
     .then((payload) => {
+      if (currentMembershipType === MEMBERSHIP_TYPE_NONE) {
+        dispatch(updateProfile(profile));
+        dispatch(updateIDPProfile(idpProfile));
+      }
       dispatch(stopLoading());
```

Comment:
add a visual feedback for a successful operation 
a modal message would do the trick